awsagentprovider/src/test/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGeneratorTest.java

```diff
@@ -1029,6 +1030,84 @@ private void validateHttpStatusForNonLocalRootWithThrowableForClient(
     }
   }
 
+  @Test
+  public void testDBUserAttribute() {
+    mockAttribute(DB_SYSTEM, "db_system");
+    mockAttribute(DB_OPERATION, "db_operation");
+    mockAttribute(DB_STATEMENT, "db_statement");
+    mockAttribute(DB_USER, "db_user");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_OPERATION)).isEqualTo("db_operation");
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isEqualTo("db_user");
+  }
+
+  @Test
+  public void testDBUserAttributeAbsent() {
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isNull();
+  }
+
+  @Test
+  public void testDBUserAttributeWithDifferentValues() {
+    mockAttribute(DB_SYSTEM, "db_system");
+    mockAttribute(DB_OPERATION, "db_operation");
+    mockAttribute(DB_STATEMENT, "db_statement");
+    mockAttribute(DB_USER, "non_db_user");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isEqualTo("non_db_user");
+  }
+
+  @Test
+  public void testDBUserAttributeNotPresentInServiceMetricForServerSpan() {
+    String dbUser = "db_user";
```

Comment:
You are testing `DBUserAttributeNotPresentInServiceMetricForServerSpanAndIsDbSpanFalse`, which is not as helpful:
```suggestion
    mockAttribute(DB_SYSTEM, "db_system");
    String dbUser = "db_user";
```